packages/eas-cli/src/metadata/config.ts

```diff
@@ -12,6 +18,81 @@ export interface MetadataConfig {
   apple?: AppleMetadata;
 }
 
+/**
+ * Resolve the dynamic config from the user.
+ * It supports methods, async methods, or objects (json).
+ */
+async function resolveDynamicConfigAsync(configFile: string): Promise<unknown> {
+  const userConfigContent = await import(configFile);
+  const userConfig = userConfigContent.default || userConfigContent;
+
+  if (typeof userConfig === 'function') {
+    return await userConfig();
+  } else if (typeof userConfig === 'object') {
+    return userConfig;
+  }
```

Comment:
@EvanBacon We can also add more "side wheels" to help users when they return anything other than an `object` from the (async) function? But not sure if necessary./

Reply:
Screw it, did it anyways.